block_issuer_key_ed25519_address.go

```diff
@@ -1,32 +1,37 @@
 package iotago
 
+import (
+	"context"
+
+	"github.com/iotaledger/hive.go/lo"
+	"github.com/iotaledger/hive.go/serializer/v2"
+)
+
 // An Ed25519 Address Block Issuer Key.
-type BlockIssuerKeyEd25519Address struct {
+type Ed25519AddressBlockIssuerKey struct {
 	Address *Ed25519Address `serix:"0"`
 }
 
-// BlockIssuerKeyEd25519FromPublicKey creates a block issuer key from an Ed25519 public key.
-func BlockIssuerKeyEd25519AddressFromAddress(address *Ed25519Address) BlockIssuerKeyEd25519Address {
-	return BlockIssuerKeyEd25519Address{Address: address}
+// Ed25519AddressBlockIssuerKeyFromAddress creates a block issuer key from an Ed25519 address.
+func Ed25519AddressBlockIssuerKeyFromAddress(address *Ed25519Address) Ed25519AddressBlockIssuerKey {
+	return Ed25519AddressBlockIssuerKey{Address: address}
 }
 
 // BlockIssuerKeyBytes returns a byte slice consisting of the type prefix and the raw address.
-func (key BlockIssuerKeyEd25519Address) BlockIssuerKeyBytes() []byte {
-	blockIssuerKeyBytes := make([]byte, 0, key.Size())
-	blockIssuerKeyBytes = append(blockIssuerKeyBytes, byte(Ed25519BlockIssuerKeyAddress))
-	return append(blockIssuerKeyBytes, key.Address[:]...)
+func (key Ed25519AddressBlockIssuerKey) BlockIssuerKeyBytes() []byte {
+	return lo.PanicOnErr(CommonSerixAPI().Encode(context.TODO(), key))
 }
 
 // Type returns the BlockIssuerKeyType.
-func (key BlockIssuerKeyEd25519Address) Type() BlockIssuerKeyType {
-	return Ed25519BlockIssuerKeyAddress
+func (key Ed25519AddressBlockIssuerKey) Type() BlockIssuerKeyType {
+	return BlockIssuerKeyEd25519Address
 }
 
 // Size returns the size of the block issuer key when serialized.
-func (key BlockIssuerKeyEd25519Address) Size() int {
-	return key.Address.Size()
+func (key Ed25519AddressBlockIssuerKey) Size() int {
+	return serializer.SmallTypeDenotationByteSize + key.Address.Size()
 }
 
-func (key BlockIssuerKeyEd25519Address) VBytes(rentStructure *RentStructure, vbyteFunc VBytesFunc) VBytes {
-	return key.Address.VBytes(rentStructure, vbyteFunc)
+func (key Ed25519AddressBlockIssuerKey) VBytes(rentStructure *RentStructure, vbyteFunc VBytesFunc) VBytes {
+	return rentStructure.VBFactorData.Multiply(VBytes(serializer.SmallTypeDenotationByteSize)) + key.Address.VBytes(rentStructure, vbyteFunc)
```

Comment:
Block Issuer Keys have their own rent structure weight, so we shouldn't use the data factor. That is currently being applied in the `BlockIssuerKeys` type, but it would be clearer if it was here, actually.

I think this should be:

```go
return rentStructure.VBFactorBlockIssuer.Multiply(VBytes(key.Size()))
```

We shouldn't delegate to `VBytes` of the address, because the block issuer key and the address type have different storage deposit rules, hence why `key.Size()` is better I think.